mantidimaging/gui/windows/operations/presenter.py

```diff
@@ -120,19 +127,54 @@ def do_apply_filter_to_all(self):
         if not confirmed:
             return
         stacks = self.main_window.get_all_stack_visualisers()
+        if self.view.safeApply.isChecked():
+            self.original_images_stack = []
+            for stack in stacks:
+                self.original_images_stack.append((stack.presenter.images.copy(), stack.uuid))
 
+        if len(stacks) > 0:
+            self.applying_to_all = True
         self._do_apply_filter(stacks)
 
-    def _post_filter(self, updated_stacks, task):
+    def _wait_for_stack_choice(self, new_stack, stack_uuid):
+        stack_choice = StackChoicePresenter(self.original_images_stack, new_stack, self, stack_uuid)
+        stack_choice.show()
+
+        while not stack_choice.done:
+            QApplication.processEvents()
+            QApplication.sendPostedEvents()
+            sleep(0.05)
+
+        return stack_choice.use_new_data
 
+    def is_a_proj180deg(self, stack_to_check):
+        if stack_to_check.presenter.images.has_proj180deg():
+            return False
+        stacks = self.main_window.get_all_stack_visualisers()
+        for stack in stacks:
+            if stack.presenter.images.proj180deg == stack_to_check.presenter.images:
+                return True
+        return False
+
+    def _post_filter(self, updated_stacks, task):
+        do_180deg = True
         for stack in updated_stacks:
+            is_a_proj180deg = self.is_a_proj180deg(stack)
+            if self.view.safeApply.isChecked() and not is_a_proj180deg:
+                do_180deg = self._wait_for_stack_choice(stack.presenter.images, stack.uuid)
```

Comment:
This should check if the task failed before showing the comparison (and before applying to the 180 degree actually), because currently if you do:

- Load some data that you can copy for safe apply
- Rebin to 9999 9999
  - The goal is to get the copy to succeed - but the filter allocation to fail

I got:
- Comparison with an unchanged stack (as the filter failed)
- Proj 180 was just gone

It should probably just show an error and pretend the user clicked "keep original"

Reply:
Should be sorted now I think :+1: